### ISSUE TITLE
Atrofac closes occasionally

### ISSUE THREAD

**Issue by LoganDark** [cronosun/atrofac#22]
Note, I say closes because of crashes, because the icon disappears the next time you hover over it, but the atrofac-gui.exe process keeps running indefinitely.

I keep atrofac open for weeks or months at a time since my computer is used at home and almost never shuts down or reboots. It's the only program that lets me control my laptop's fans (ROG Zephyrus M15) without the stupid bloated "Armoury Crate", which takes 3 loading screens and 30 seconds to switch the fan speed (and is also British English, by the way, which has NO PLACE on a US machine).

My only issue with atrofac is this occasional closing. It can be easily restarted with no issue but of course the next time I open task manager there are 15 atrofacs due to how many disappeared their taskbar icon but kept running somehow.

**Comment by dnmTX** [comment]
@LoganDark by any chance your M15 is the 4K 60Hz version?

**Comment by LoganDark** [comment]
> @LoganDark by any chance your M15 is the 4K 60Hz version?

Yes, although I severely regret going with this particular 4K laptop. It has the worst color issues and ghosting that I've ever seen on a display.

Does the 4K version have some sort of unique issue that affects atrofac?

**Comment by dnmTX** [comment]
> Does the 4K version have some sort of unique issue that affects atrofac?

Not that i know off and i'm off topic here,sorry if my question misled you.

> Yes, although I severely regret going with this particular 4K laptop. It has the worst color issues and ghosting that I've ever seen on a display.

By any chance you still have the original color profile that came when the laptop was still in stock?
Same question about the original theme.

**Comment by LoganDark** [comment]
> By any chance you still have the original color profile that came when the laptop was still in stock?

Yes. You can place these files in `C:\Windows\System32\spool\drivers\color`: [GU502LV_8086_AF0631EB_CMDEF.zip](https://github.com/cronosun/atrofac/files/8009626/GU502LV_8086_AF0631EB_CMDEF.zip)

But I will warn you, it won't help with the insane color issues that this display has. Your dedicated GPU will still have incorrect colors, the display hardware will still wildly oversaturate everything, and Windows will still not recognize the display as WCG.

> Same question about the original theme.

Place the files inside this zip in `%localappdata%\Microsoft\Windows\Themes`: [oem.zip](https://github.com/cronosun/atrofac/files/8009631/oem.zip)

And this in `%SystemRoot%\ASUS\Wallpapers`:

![ASUS](https://user-images.githubusercontent.com/4723091/152670989-ef06300b-78c7-4f37-ab29-6af2e6cffb30.jpg)

In fact here's the entire `%SystemRoot%\ASUS` folder, zipped: [ASUS.zip](https://github.com/cronosun/atrofac/files/8009633/ASUS.zip)

**Comment by dnmTX** [comment]
Nice,very nice. Thank you 👍 
And yeah,the display is...pffff...wont even comment. Response time is 25.6 ms,compare to any other 1080p from the same model: 8.8 ms. So ghosting is probably at least of our worries here 😄 
You can read very detailed review here,display is crappy,indeed. I wonder,down the road if it will be possible to replace it with some QHD(2560x1440) one.
https://www.notebookcheck.net/Asus-ROG-Zephyrus-M15-GU502L-in-review-Compact-gaming-notebook-with-Turbo-reserves.496316.0.html

**Comment by dnmTX** [comment]
@LoganDark there is a `PQConfig.dv` with the color profile. Any idea what is for,never seen such a file before.

**Comment by LoganDark** [comment]
> I wonder,down the road if it will be possible to replace it with some QHD(2560x1440) one.

That won't help with the color space issues. The built-in integrated graphics are somehow configured (in hardware) to color-correct the RTX GPU so that the colors look accurate on the display. Problem is that the color correction shows up in screenshots. Disgusting.

Not to mention QHD displays are significantly lower res than 4K and I wouldn't be able to comfortably use one at an integer scale factor. Which is a big problem considering non-integer is a hot mess.

> @LoganDark there is a `PQConfig.dv` with the color profile. Any idea what is for,never seen such a file before.

No idea what it is either. It was in the folder and had the same modification date as the factory color profile so I included it in the zip for completeness. It's probably something they used for the color calibration.

**Comment by dnmTX** [comment]
I just did little research and it turns out it's needed for Dolby Vision,so thanks,i didn't have that one either 👍

**Comment by dnmTX** [comment]
Any tips on the fan profiles. At the moment i'm using the CLI version with `plan performance` but i really want to make it a bit more aggressive so to say. The thing is that no one so far can provide info on how the fan curves are set up on any of the plans but the silent one.

**Comment by LoganDark** [comment]
> Any tips on the fan profiles. At the moment i'm using the CLI version with `plan performance` but i really want to make it a bit more aggressive so to say. The thing is that no one so far can provide info on how the fan curves are set up on any of the plans but the silent one.

Performance runs the fans at >100%. It's not possible to get faster than that.

My laptop only has 1 functional fan (the GPU fan) and it still doesn't thermal throttle under Performance unless the intake is blocked (by blanket or etc). It gets hotter than an Intel Mac though.

**Comment by dnmTX** [comment]
> Performance runs the fans at >100%. It's not possible to get faster than that.

Nope,the `turbo` is the one that does that. The `performance` is very close to the stock windows one but still not good enough for me though 😉 

> My laptop only has 1 functional fan (the GPU fan)

How is that even possible? Unless your other one broke or something?

**Comment by LoganDark** [comment]
> Nope,the `turbo` is the one that does that.

Wrong, performance is faster than turbo. Turbo is a constant 100% while performance has the ability to have short bursts of >100%.

> How is that even possible? Unless your other one broke or something?

The laptop fell and both fan bearings instantly went bad. The CPU fan (left fan) was way more annoying than the GPU fan (right fan), so I jammed the worse one and have to deal with the GPU fan randomly entering hyperdrive and needing to be restarted (which is admittedly very easy due to the laptop's shitty plastic construction). I have new fans and thermal paste on order from eBay.

**Comment by dnmTX** [comment]
So ASUS used 1080p wallpaper on laptop with 4K display. It's another **NO COMMENT** from me 😄

**Comment by LoganDark** [comment]
> So ASUS used 1080p wallpaper on laptop with 4K display. It's another **NO COMMENT** from me 😄

This entire laptop is full of questionable choices, i.e. they went with another NVMe slot instead of another RAM slot. I have 40GB of memory and only the first 16GB is dual channel. Because one of the RAM sticks are soldered to the board. Realistically they could have made the laptop like 2 mm thicker for a common double-stacked memory bay but they didn't because of the stupid fad. Also down firing speaker is the stupidest thing I've ever heard, literally. Just goes into the blanket and gets absorbed. Luckily I have my own USB sound system.

**Comment by dnmTX** [comment]
> I have new fans and thermal paste on order from eBay

I was about to suggest that as i saw that they sell the fans there in the past.
About repasting.....well prep yourself for a big fight and try not to damage the cooling pipes while removing them.
They used that blue goo insted of thermal pads and that thing is like welded. Trust me,if you're not mentally ready,better don't even start 😄 
Would highly advise to go with LM on the CPU though.
Just to return the favor,as you helped me alot with providing the color profile and the rest(BIG BIG THANKS AGAIN 👍 ) here is what you're going to be dealing with 👇  : 

![cooling pipes(before)](https://user-images.githubusercontent.com/34774426/152672221-d33321ae-46eb-436c-b4bb-003a30f8de33.jpg)

![main board(before)](https://user-images.githubusercontent.com/34774426/152672225-df770daf-d13a-41a3-a43f-252f95005f06.jpg)

**Comment by LoganDark** [comment]
> About repasting.....well prep yourself for a big fight and try not to damage the cooling pipes while removing them.
> They used that blue goo insted of thermal pads and that thing is like welded. Trust me,if you're not mentally ready,better dont't even start 😄

Kinda don't have any choice here. This laptop needs to survive for about another year until I have enough money to REALLY splurge on a quality system, which costs about $6,000 these days (from the likes of Clevo Computer). These fans are going to die. One already doesn't work properly. The other one hates whenever I slightly rotate the system. And in order to replace the fans you have to remove the pipes due to the connectors and ducts being underneath them.

Plus neutering windows is extremely extremely annoying. So many backdoors and telemetry. Barely managed to keep Windows Mixed Reality working through all the ripping out of "Critical" system components I had to do, considering you can't connect the machine to the internet or else it will try to update itself. So I'd like to do that as infrequently as possible.

> Would highly advise to go with LM on the CPU though.

I only have one shot at this, I'm not LM'ing my daily driver laptop. Especially since if the CPU or motherboard or anything dies due to an LM short I am 100% screwed. I don't have any skill with this. I'm used to the devices I buy being actually good from the factory. Apple used to be great at that.

Thanks for the pictures of the cooling system and for the warning.

**Comment by dnmTX** [comment]
> I'm used to the devices I buy being actually good from the factory

Well....as you see from the pictures(and this is the case with all of them M15's G15's G14's and so on,and so on) they did very shitty job on the cooling,especially with the LM and not to mention the paste on the GPU was extremely dry.
Good luck with the project,i know from my experience that you'll need it 👍 
Thanks again for helping me,appreciated very much ❤️ 
Oh....and i honestly wouldn't hold my breath for response or any assist for that matter from the repo owner(developer) here.
Looks like that project has been abandoned,which is sad.
My advise,if you're not using any custom fan curves but the stock ones,just go with the CLI version and create a schedule task for it and be done with it.
If you need any help with that,let me know,would gladly help you 👍

**Comment by LoganDark** [comment]
@dnmTX I went ahead with the repair. Took off the heatsink assembly, scraped off all the OEM thermal paste, cleaned all surfaces with isopropyl alcohol (except the CPU). Re-spread the LM on the CPU, applied Arctic Silver Ceramique thermal paste to all the places where the "blue goo" used to be, put the heatsink assembly back on.

So yeah I kept the LM on the CPU. It came from the factory with so much LM that all it needed was a re-application with a qtip. Although I knew for a fact it didn't have enough to transfer onto the GPU as well so I just used thermal paste for that.

LM on cpu

![image](https://user-images.githubusercontent.com/4723091/153750039-9615e215-4d4e-4538-9743-eead21cd4ba7.png)

Thermal paste application (my first time ever):

![image](https://user-images.githubusercontent.com/4723091/153750062-425874bb-982a-4149-a5a2-fce90bdcbe0a.png)

Thanks for your advice and the images! If you ever need a fan replacement, go with these, they are 12V and fit just fine: https://www.ebay.com/itm/254670490532
